src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -235,6 +236,9 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
 
         auto pass_config = manager.get_pass_config();
         pass_config->disable<ov::pass::EyeDecomposition>();
+
+        // in order to keep ShapeOf precision sensitive subgraphs in FP32
```

Comment:
Need better comment that this transformation exist in CommonOptimizations and in Convert to legacy